en/sensor/pmw3901.md

```diff
@@ -0,0 +1,76 @@
+<span id="Thone"></span>[Thone](https://www.seeedstudio.com/ThoneFlow-3901U-UART-Serial-Version-PMW3901-Optical-Flow-Sensor-p-4040.html) | UART | X | - | - | 3 - 5 | AxB
+
+## Specifications
+
+* TODO
+* TODO
+* TODO
+
+## Pixhawk Setup
+
+### Bitcraze
+
+The Bitcraze optical flow module exposes directly the SPI interface from the PMW3901 module. The board also has a distance sensor that can be wired via I2C. This distance sensor is the VL53L0x ToF sensor for STMicroelectronics. The range of the sensor is very limited (2 meters) and will be reduced when flying in the sunlight. We therefore highly recommend to use another external distance sensor. This board was mainly tested with the [Lidar Lite V3](../sensor/lidar_lite.md) on larger vehicles and with the [Lanbao CM8JL65](../sensor/cm8jl65_ir_distance_sensor.md) on smaller vehicles. 
+
+The PMW3901 driver was written to be directly plugged into the SPI port on the Pixhawk 4 using the chip select 1. The support for this driver on other boards can then easily be exteded to other boards by modifying the SPI_BUS and the SPI_DEV in the PMW3901 driver. The following diagram shows how to wire the sensor to a Pixhawk 4.
+
+![Bitcraze PH4 Pinout](../../assets/hardware/sensors/pmw3901/ph4-bitcraze-flow-pinout.png)
+
+When plugged into the SPI port of the Pixhawk 4, the Bitcraze flow module will be automatically detected at bootup. No parameters will have to configured, unless you decide to change the orientation of the sensor. This will be explained later in this section together with some diagrams to show the default orientation of the sensor.
+
+### Tindie
+
+The Tindie optical flow module exposes the SPI interface from the PMW3901 module exactly as on the Bitcraze module. The only difference here is the size of the board and the fact that this module doesn't have a distance sensor onboard. Unlike the Bitcraze module where you could try to use the onboard distance sensor, with this one it is mandatory to use one of the [distance sensors](../sensor/rangefinders.md) supported by PX4. 
+
+![Tindie PH4 Pinout](../../assets/hardware/sensors/pmw3901/ph4-tindie-flow-pinout.png)
+
+### Hex
+
+The Hex optical flow module is a very small board containing the PMW3901 flow module, the VL53L1X distance sensor and an IMU to better synchronize the flow data with the gyro data. These three sensors are then sampled by a microcontroller onboard. The goal of the microcontroller is to then expose 2 UAVCAN messages containing all the necessary data.
+
+This board can be connected to the CAN port on any Pixhawk board. All tests were performed connecting it to the CAN1 port on a Pixhawk 4. Differently from the other sensors, this one needs UAVCAN running on PX4. To enable UAVCAN you will have to set the UAVCAN_ENABLE parameter to 2 (if you have have UAVCAN sensors but no motor controllers) or to 3 (if you have also your motor controllers on CAN).
+
```

Comment:
If I use an external distance sensor, is there anything special I need to do to make sure that PX4 properly fuses the external one rather than using the inbuilt one that comes with UAVCAN message?

Reply:
not sure anymore. I need to check this one